openvdb_ax/openvdb_ax/compiler/AttributeBindings.h

```diff
@@ -0,0 +1,158 @@
+// Copyright Contributors to the OpenVDB Project
+// SPDX-License-Identifier: MPL-2.0/
+
+/// @file compiler/AttributeBindings.h
+///
+/// @authors Richard Jones
+///
+/// @brief The Attribute Bindings class is used by the compiled Executables
+///   to handle the mapping of AX Attribute names to context dependent data
+///   names, i.e point attribute and volume grid names.
+///
+
+#ifndef OPENVDB_AX_COMPILER_ATTRIBUTE_BINDINGS_HAS_BEEN_INCLUDED
+#define OPENVDB_AX_COMPILER_ATTRIBUTE_BINDINGS_HAS_BEEN_INCLUDED
+
+#include <openvdb/version.h>
+#include <openvdb/util/Name.h>
+
+#include <map>
+
+namespace openvdb {
+OPENVDB_USE_VERSION_NAMESPACE
+namespace OPENVDB_VERSION_NAME {
+
+namespace ax {
+
+/// @brief This class wraps an interface for a map of attribute bindings. These map
+///   attributes in AX code to context data. These mappings are one-to-one i.e.
+///   each AX name can only map to one data name, however each name can appear as either
+///   an AX name or data name or both, i.e. the following sets of bindings are valid:
+///   axname: a   ->  dataname: a
+///   axname: b   ->  dataname: c
+///   or
+///   axname: a   ->  dataname: b
+///   axname: b   ->  dataname: a
+class AttributeBindings
+{
+public:
+
+    AttributeBindings() {}
+
+    AttributeBindings(const AttributeBindings& other)
+        :  mAXToDataMap(other.mAXToDataMap), mDataToAXMap(other.mDataToAXMap) {}
```

Comment:
I think we can remove this copy constructor and let the compiler init it. Do we need the empty constructor too? i.e. is this class ever created without a bindings vector?

Reply:
in the sop it currently does create a bindings vector and then populate it, this could obviously be changed to use some temporary vector if needed. I can remove these though and let the compiler do it for me.